blog/content/post/2024-08-07-odroid-h4-analysis.md

```diff
@@ -0,0 +1,538 @@
+  Controller, USB-PD, TouchPads, etc. One must know what their address is to
+  properly implement support for them.
+- `inputbustypes` - these files represent the content of
+  `/sys/class/input/input*/id/bustype`. not sure what it is used for. It was
+  originally included in [coreboot's mainboard porting
+  guide](https://www.coreboot.org/Motherboard_Porting_Guide)
+- `intelmetool` - queries information about ME and BootGuard, if ME device is
+  supported by the utility, it may give useful information whether Boot Guard
+  is fused or not. This coreboot's utility is not updated often and worked
+  well on rather older platforms, like 7th gen Intel Core and older.
+- `ioports` - very useful to get information about I/O port usage. One may
+  find out which port ranges are used and by which devices, for example, which
+  I/O port ranges are used by Super I/O.
+- `lspci` - also very useful file containing a complete dump of PCI
+  configuration space of the system. I often used it to compare the PCI/PCIe
+  capabilities of devices, differences in PCI configuration space to look for
+  mistakes in open-source firmware settings for PCIe initialization or simply
+  to get the PCI Subsystem IDs used by the platforms.
+- `lsusb` - useful to map the USB ports, plugged and on-board devices. WHen
+  you don't have the schematics invoking `usb -t` and plugging devices in all
+  ports helps to map the used USB ports. But we have a schematics here, so it
+  usefulness is not that big.
+- `msrtool` - another coreboot's utility which was implemented for older
+  platforms, for modern ones it does not dump anything. Rather unmaintained.
+- `nvramtool` - dumps the CMOS NVRAM space and can modify the CMOS NVRAM
+  options if CMOS option backed is enabled in coreboot. Sometimes it may be
+  useful if the firmware stores some information in CMOS. Dumping it across
+  boots may reveal some BIOS implementation details.
+- `result` - a file with results of dumping each information type from the
+  system, integral part of HCL Report, not used for porting
+- `superiotool` - another crucial file. If you system has a Super I/O chip,
+  like ODROID, it dumps the configuration registers of said chip. If you don;t
+  have schematics for the platform, you are probably doomed without the log
+  from superiotool.
+- `touchpad` - it is a result of a
+  [script](https://github.com/Dasharo/meta-dts/blob/main/meta-dts-distro/recipes-dts/reports/touchpad-info/touchpad-info),
+  which tries to probe information about touchpads connected to I2C. Rather
+  useful on laptops, not board like ODROID.
+- `tpm_version` - contains information about TPM module read from
+  `/sys/class/tpm/tpm*`. Tell us if any TPM module is connected/used on the
+  board. ODROID has no discrete TPM module, but it can still use fTPM.
+
+I will now go through the most important files and explain why we need it and
+how to use it,
+
+### SMBIOS information
+
+TBD
```

Comment:
Any plans to fill these in?

Reply:
Yes, when I find time to finish the post